test/leafTest.js

```diff
@@ -0,0 +1,151 @@
+const { BigNumber } = require("ethers")
+const { expect } = require("chai")
+
+describe("Leaf", () => {
+  let leafInstance
+
+  beforeEach(async () => {
+    const LeafStub = await ethers.getContractFactory("LeafStub")
+    leafInstance = await LeafStub.deploy()
+    await leafInstance.deployed()
+  })
+
+  describe("make", async () => {
+    it("constructs a leaf", async () => {
+      // We store a leaf as a uint256, which is 256 bits. We pack the address
+      // in the first 160 bits, the creation block in the 64 bits after that,
+      // and the operator's id in the final 32 bits. Each hex character
+      // represents 4 bits, so a 40-character hex representation of an eth
+      // address is the exact 160 bits we need. A number (like a creation block
+      // or id) gets left-padded with zeros to fill the exact bit count.
+      const testData = [
+        {
+          address: "0x4ddc2d193948926d02f9b1fe9e1daa0718270ed5",
+          creationBlock: BigNumber.from("0xDFD871"),
+          id: 0x1995b5,
+          expectation:
+            "0x4ddc2d193948926d02f9b1fe9e1daa0718270ed50000000000DFD871001995B5",
+          //     read:    4ddc2d193948926d02f9b1fe9e1daa0718270ed5,0000000000DFD871,001995B5
+          //                    address                             creationBlock     id
+        },
+        {
+          address: "0x176f3dab24a159341c0509bb36b833e7fdd0a132",
+          creationBlock: BigNumber.from("0x5E8EEE0"),
+          id: 0x8f573,
+          expectation:
+            "0x176f3dab24a159341c0509bb36b833e7fdd0a1320000000005E8EEE00008F573",
+          //     read:    176f3dab24a159341c0509bb36b833e7fdd0a132,0000000005E8EEE0,0008F573
+          //                    address                             creationBlock     id
+        },
+        {
+          address: "0x2b6ed29a95753c3ad948348e3e7b1a251080ffb9",
+          creationBlock: BigNumber.from("0x771D8DB"),
+          id: 0xca009,
+          expectation:
+            "0x2b6ed29a95753c3ad948348e3e7b1a251080ffb9000000000771D8DB000CA009",
+          //     read:    2b6ed29a95753c3ad948348e3e7b1a251080ffb9,000000000771D8DB,000CA009
+          //                    address                             creationBlock     id
+        },
+      ]
+
+      testData.forEach(async (test) => {
+        const leaf = await leafInstance.make(
+          test.address,
+          test.creationBlock,
+          test.id,
+        )
+        expect(leaf).to.equal(test.expectation)
```

Comment:
Worth adding an error message indicating which index failed (`forEach` injects the index argument so it is accessible when we do `async (test, index) => ...`). Same for other `expect` living in `forEach` loops in this test suite.

Reply:
[aba71d3](https://github.com/keep-network/sortition-pools/pull/173/commits/aba71d3490742cc92485bed748a265721759dbde)

Reply:
Looks like https://github.com/keep-network/sortition-pools/pull/173/commits/aba71d3490742cc92485bed748a265721759dbde has a typo `text` instead of `test`?

Reply:
[a6d39b0](https://github.com/keep-network/sortition-pools/pull/173/commits/a6d39b07dd683d62a652e67da307995aebb6b366)